src/encoded/types/__init__.py

```diff
@@ -247,7 +247,7 @@ def unique_keys(self, properties):
         "type": "string",
     })
     def publication_year(self, date_published):
-        return date_published.partition(' ')[0]
+        return int(date_published[:4])
```

Comment:
I see that you check whether the value can be coerced in the upgrade to a number... but it's not checked here.  You probably need a try/except to prevent this from crashing if somehow date_published doesn't match the regex... On the other hand ... I guess it would return 0 as a valid non-year INT?

Reply:
maybe it can just return None if it cannot be cast as int.

Reply:
I added the same check as in the upgrade and had it return None explicitly if it failed the check